### DISCUSSION TITLE
Array with required item

### DISCUSSION THREAD

**Discussion by HonzaKopecky** [colinhacks/zod#3002] [Q&A] [answered]
According to [this](https://github.com/colinhacks/zod/discussions/2278), we can specify the first element of an array to be an exact value using `tuple()` but how would I go to define a required element(s)? I don't care about the position in the array but I need the element(s) to be present there.

**Comment by JacobWeisenburger** [comment]
Perhaps this will help
https://github.com/colinhacks/zod#tuples

**Reply by HonzaKopecky** [reply]
Well, to me this seems like it requires the position of the items to be exactly as defined which is not what I am trying to achieve.

**Comment by JacobWeisenburger** [comment, marked as answer]
Please provide some example data that you are trying to validate.
